src/plugins/navigation/server/ui_settings.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { CoreSetup, KibanaRequest } from '@kbn/core/server';
+import { schema } from '@kbn/config-schema';
+import { UiSettingsParams } from '@kbn/core/types';
+import { i18n } from '@kbn/i18n';
+import { isRelativeUrl } from '@kbn/std';
+import type { Space } from '@kbn/spaces-plugin/server';
+
+import { DEFAULT_ROUTE_UI_SETTING_ID, DEFAULT_ROUTES } from '../common/constants';
+import { NavigationServerStartDependencies } from './types';
+
+const cache: { request?: KibanaRequest; activeSpace?: Space } = {};
+
+/**
+ * uiSettings definitions for Navigation
+ */
+export const getUiSettings = (
+  core: CoreSetup<NavigationServerStartDependencies>
+): Record<string, UiSettingsParams> => {
+  return {
+    [DEFAULT_ROUTE_UI_SETTING_ID]: {
+      name: i18n.translate('navigation.ui_settings.params.defaultRoute.defaultRouteTitle', {
+        defaultMessage: 'Default route',
+      }),
+      getValue: async ({ request }: { request?: KibanaRequest } = {}) => {
+        const [_, { spaces }] = await core.getStartServices();
+
+        if (!spaces || !request) {
+          return DEFAULT_ROUTES.classic;
+        }
+
+        try {
+          let activeSpace = cache.activeSpace;
+          if (cache.request !== request || !activeSpace) {
```

Comment:
For what optimization is needed? Do we cache only the last request?  I also wonder if there is a better place for this improvement (e.g a cache layer inside getActiveSpace)  or if it is not needed altogether, as I wonder if there is not much benefit to it if multiple users are using Kibana?

Reply:
Ok let's remove it and add only add it if needed later as an optimisation 👍